### DISCUSSION TITLE
[Idea] Redis

### DISCUSSION THREAD

**Discussion by DracoBluePL** [PrestaShop/PrestaShop#35546] [Ideas]
### The problem

I'm always frustrated when PrestaShop, like many other e-commerce platforms, struggles with handling high volumes of data and concurrent user requests efficiently. This often leads to slow loading times, which can significantly impact user experience and ultimately, sales.

### The solution

Integrating Redis into PrestaShop's architecture can significantly alleviate these performance issues. Redis is an in-memory data store that can serve as a caching layer, reducing the need for repeated database queries and speeding up data retrieval. By caching frequently accessed data such as product information, session data, and database query results, Redis can dramatically improve response times and overall system performance.

Changing to Redis would help because:

**Faster response times:** Redis stores data in memory, allowing for lightning-fast retrieval compared to traditional disk-based databases. This means pages load quicker, leading to a smoother shopping experience for customers.

**Scalability:** Redis is highly scalable and can handle increasing loads of data and traffic without sacrificing performance. This is crucial for growing e-commerce businesses that need to accommodate spikes in traffic, especially during peak seasons or promotional events.

**Improved user experience:** With Redis caching frequently accessed data, customers experience faster page load times and smoother navigation throughout the site. This enhances user satisfaction and encourages repeat visits and purchases.

### Alternatives

While there are other approaches to improving PrestaShop's performance, such as optimizing database queries, implementing a content delivery network (CDN), or upgrading hardware resources, these solutions may not provide the same level of performance improvement as Redis. Additionally, they might require more effort to implement and maintain compared to integrating Redis, which offers a straightforward caching solution.

### Additional context

Integrating Redis into PrestaShop can be done through various modules or plugins specifically designed for this purpose. These modules typically provide easy configuration options and seamless integration with PrestaShop's existing infrastructure, making it a relatively simple process for developers to implement. Additionally, Redis is open-source and widely supported, ensuring compatibility with PrestaShop's ecosystem and future updates.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
Hello @DracoBluePL I think the issue https://github.com/PrestaShop/PrestaShop/issues/26129 will provide you some answers 👍

**Comment by kpodemski** [comment]
There are great modules that handle Redis support in PrestaShop. There is also a PageCache module that does the job well, and it doesn't use Redis. In other words: it's a problem, but there are solutions available, if you have high traffic, 200EUR for a module shouldn't be an issue, right?

**Reply by DracoBluePL** [reply]
As we can see, PrestaShop relies on outdated cache systems. It would be worthwhile to remove these obsolete cache systems and replace them with modern ones. Unfortunately, in my opinion, such a module should be managed by PrestaShop rather than commercial providers.

**Reply by matks** [reply]
Integrating Redis into PrestaShop would be a massive work needing multiple developers for a long time. As developer time is not free, I think the only way to make it happen is through a paid module or service.

The scenario where a company (PrestaShop or another) pays multiple developers for multiple weeks to build Redis feature into PrestaShop for free does not work economically. And a company does not do things that are not a good ROI.
